### PR TITLE
[AMORO-2875] Added metric for orphan file action

### LINKED ISSUE #2875: [Improvement]: Action `clean-orphan-files`  should report metric
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Action `clean-orphan-files`  should report metric

For tables that generate a lot of orphan files, we need to check whether the writing task is normal. This indicator is very helpful in this scenario.

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by huyuanfeng2018**
## Why are the changes needed?

Close #2875

## Brief change log

- added `TableOrphanFilesCleaningMetrics` used to collect metrics generated by cleaning orphan files

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Review by XBaith — COMMENTED** [review]
we can try to collect more detailed information, such as the size of deleted files

File: docs/user-guides/metrics.md [hunk 1]
```diff
@@ -60,13 +64,22 @@ Amoro has supported built-in metrics to measure status of table self-optimizing
...
+| optimizer_group_threads                | Gauge | group | Number of total threads in optimizer group       |
+
+## Orphan Files Cleaning metrics
+
+| Metric Name                                      | Type    | Tags                     | Description                                                                  |
+|--------------------------------------------------|---------|--------------------------|------------------------------------------------------------------------------|
+| table_orphan_content_cleaning_count              | Counter | catalog, database, table | Count of orphan content files cleaned in the table since ams started         |
+| table_orphan_metadata_file_cleaning_count        | Counter | catalog, database, table | Count of orphan metadata files cleaned in the table since ams started        |
+| table_slated_orphan_content_file_cleaning_count  | Counter | catalog, database, table | Slated Count of orphan content files cleaned in the table since ams started  |
+| table_slated_orphan_metadata_file_cleaning_count | Counter | catalog, database, table | Slated Count of orphan metadata files cleaned in the table since ams started |
```
Comment (baiyangtx) [inline docs/user-guides/metrics.md:85]: What does `slated_orphan_content_file` and `slated_orpha_metadata_file` means?

Comment (huyuanfeng2018) [reply docs/user-guides/metrics.md:85]: Indicates the calculated number of files expected to be deleted. Failure may occur during the deletion process, so there will also be an metric of the actual number of deleted files.

**Comment by huyuanfeng2018** [comment]
@XBaith

**Comment by huyuanfeng2018** [comment]
@XBaith 
[supportBulkOperations](https://iceberg.apache.org/docs/nightly/spark-procedures/#output_7)

**Comment by huyuanfeng2018** [comment]
https://iceberg.apache.org/docs/nightly/spark-procedures/#output_7
@XBaith 
It seems that iceberg only lists the deleted files and does not calculate the size. I looked at the code in Ams and found that the statistical size changes code are a bit large. I think the current file number indicator is good enough. WDYT?

**Comment by zhoujinsong** [comment]
Hi， @huyuanfeng2018   
There seem to be some conflicts that should be fixed.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  private final Counter slatedOrphanMetadataFilesCount = new Counter();
+
+  private final ServerTableIdentifier identifier;
+
+  public TableOrphanFilesCleaningMetrics(ServerTableIdentifier identifier) {
+    this.identifier = identifier;
+  }
+
+  public static final MetricDefine TABLE_ORPHAN_CONTENT_FILE_CLEANING_COUNT =
+      defineCounter("table_orphan_content_cleaning_count")
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java:50]: Is `table_orphan_content_file_cleaning_count` proper here, being consistent with the `table_orphan_metadata_file_cleanning_count` ?

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+          .build();
+
+  public static final MetricDefine TABLE_ORPHAN_METADATA_FILE_CLEANING_COUNT =
+      defineCounter("table_orphan_metadata_file_cleaning_count")
+          .withDescription("Count of orphan metadata files cleaned in the table since ams started")
+          .withTags("catalog", "database", "table")
+          .build();
+
+  public static final MetricDefine TABLE_SLATED_ORPHAN_CONTENT_FILE_CLEANING_COUNT =
+      defineCounter("table_slated_orphan_content_file_cleaning_count")
```
Comment (zhoujinsong) [inline amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java:62]: Is the name `table_expected_orphan_content_file_cleaning_count` easier to understand here?

**Review by zhoujinsong — COMMENTED** [review]
I left some small suggestion about the metric name, PTAL.

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java (hunk 2 above)
Comment (huyuanfeng2018) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java]: Well, there was a mistake here, it's been fixed

File: amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java (hunk 3 above)
Comment (huyuanfeng2018) [reply amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/table/TableOrphanFilesCleaningMetrics.java]: I originally inherited the style of the old code and named it `slated`, but now it seems `expected` is easier to understand.

**Comment by zhoujinsong** [comment]
@huyuanfeng2018 Some conflicts seem to need to be fixed.

**Comment by huyuanfeng2018** [comment]
> @huyuanfeng2018 Some conflicts seem to need to be fixed.

fixed & PTAL.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**